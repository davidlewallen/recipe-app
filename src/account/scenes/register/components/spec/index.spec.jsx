import React from 'react';
import { shallow } from 'enzyme';

import Register from '../';

describe('Register component snapshot test', () => {
  const props = {
    username: 'testUsername',
    handleUsername: () => {},
    password: 'testPassword',
    handlePassword: () => {},
    email: 'test@email.com',
    handleEmail: () => {},
    register: () => {},
    error: { value: false, message: '' },
  };

  it('should match snapshot', () => {
    const component = shallow(<Register {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('should display error messages', () => {
    const component = shallow(<Register {...props} error={{ value: true, message: 'testError' }} />);

    expect(component).toMatchSnapshot();
  });
});
