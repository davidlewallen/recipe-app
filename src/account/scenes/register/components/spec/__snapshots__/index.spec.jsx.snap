// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component snapshot test should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register
    email="test@email.com"
    error={
      Object {
        "message": "",
        "value": false,
      }
    }
    handleEmail={[Function]}
    handlePassword={[Function]}
    handleUsername={[Function]}
    password="testPassword"
    register={[Function]}
    username="testUsername"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <form
          className="register"
        >
          <input
            className="username-input"
            onChange={[Function]}
            placeholder="Username"
            value="testUsername"
          />
          <input
            className="password-input"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value="testPassword"
          />
          <input
            className="email-input"
            onChange={[Function]}
            placeholder="Email Address"
            value="test@email.com"
          />
          <button
            className="register-button"
            onClick={[Function]}
          >
            Register
          </button>
        </form>,
      ],
      "className": "register-container",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="username-input"
              onChange={[Function]}
              placeholder="Username"
              value="testUsername"
            />,
            <input
              className="password-input"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value="testPassword"
            />,
            <input
              className="email-input"
              onChange={[Function]}
              placeholder="Email Address"
              value="test@email.com"
            />,
            <button
              className="register-button"
              onClick={[Function]}
            >
              Register
            </button>,
          ],
          "className": "register",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "username-input",
              "onChange": [Function],
              "placeholder": "Username",
              "value": "testUsername",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "password-input",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "testPassword",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "email-input",
              "onChange": [Function],
              "placeholder": "Email Address",
              "value": "test@email.com",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Register",
              "className": "register-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Register",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <form
            className="register"
          >
            <input
              className="username-input"
              onChange={[Function]}
              placeholder="Username"
              value="testUsername"
            />
            <input
              className="password-input"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value="testPassword"
            />
            <input
              className="email-input"
              onChange={[Function]}
              placeholder="Email Address"
              value="test@email.com"
            />
            <button
              className="register-button"
              onClick={[Function]}
            >
              Register
            </button>
          </form>,
        ],
        "className": "register-container",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="username-input"
                onChange={[Function]}
                placeholder="Username"
                value="testUsername"
              />,
              <input
                className="password-input"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value="testPassword"
              />,
              <input
                className="email-input"
                onChange={[Function]}
                placeholder="Email Address"
                value="test@email.com"
              />,
              <button
                className="register-button"
                onClick={[Function]}
              >
                Register
              </button>,
            ],
            "className": "register",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "username-input",
                "onChange": [Function],
                "placeholder": "Username",
                "value": "testUsername",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "password-input",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "testPassword",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "email-input",
                "onChange": [Function],
                "placeholder": "Email Address",
                "value": "test@email.com",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Register",
                "className": "register-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Register",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
